### PR TITLE
Support squash merges natively

### PR DESCRIPTION
**Pull request by tmeasday**
See https://github.com/chromaui/chromatic/pull/3834 / https://github.com/chromaui/chromatic/issue/1132

### CODE REVIEW

File: bin/git/git.js [hunk 1]
```diff
@@ -208,15 +213,16 @@ async function step(
...
   // Include the latest build from this branch as an ancestor of the current build
   const {
-    app: { firstBuild, lastBuild },
+    app: { firstBuild, lastBuild, pullRequest },
   } = await client.runQuery(TesterFirstCommittedAtQuery, {
     branch,
+    commit,
   });
-  debug(`App firstBuild: ${firstBuild}, lastBuild: ${lastBuild}`);
+  debug(`App firstBuild: ${firstBuild}, lastBuild: ${lastBuild}, pullRequest: ${pullRequest}`);
```
Comment (ghengeveld) [inline bin/git/git.js:225]: Probably this is just going to yield `[object Object]`. Same goes for the first/last build of course.

File: bin/git/git.js [hunk 2]
```diff
@@ -226,22 +232,43 @@ export async function getBaselineCommits(client, { branch, ignoreLastBuildOnBran
...
+  // [Don't do any special branching logic for builds on `HEAD`, this is fairly meaningless
+  // (CI systems that have been pushed tags can not set a branch)]
   if (
     branch !== 'HEAD' &&
     !ignoreLastBuildOnBranch &&
     lastBuild &&
     lastBuild.committedAt <= committedAt
   ) {
     if (await commitExists(lastBuild.commit)) {
+      debug(`Adding last branch build commit ${lastBuild.commit} to commits with builds`);
```
Comment (ghengeveld) [inline bin/git/git.js:246]: Did you test these debug statements? I'm not sure setupDebug still works. I typically use `ctx.log.debug()`.

File: bin/git/git.js [hunk 3]
```diff
@@ -226,22 +232,43 @@ export async function getBaselineCommits(client, { branch, ignoreLastBuildOnBran
   const initialCommitsWithBuilds = [];
   const extraBaselineCommits = [];
 
-  // Don't do any special branching logic for builds on `HEAD`, this is fairly meaningless
-  // (CI systems that have been pushed tags can not set a branch)
+  // Add the most recent build on the branch as a baseline (unless the user opts out), or it is newer than this commit(?)
```
Comment (ghengeveld) [inline bin/git/git.js:235]: I don't understand the 2nd half of this sentence. Can you rephrase / simplify?

File: bin/git/git.js [hunk 4]
```diff
@@ -59,6 +59,11 @@ const TesterFirstCommittedAtQuery = gql`
         commit
         committedAt
       }
+      pullRequest(mergeInfo: { commit: $commit, baseRefName: $branch }) {
+        lastHeadBuild {
+          commit
+        }
+      }
```
Comment (ghengeveld) [inline bin/git/git.js:66]: Did you expose this in `authorizedAppFieldsViaAppCode`?

Comment (tmeasday) [reply bin/git/git.js:66]: Once I tested this out I did ;)

**Comment by tmeasday** [comment]
Will merge this when https://github.com/chromaui/chromatic/issues/1132 is deployed to staging so it doesn't completely break the build.
